src/libs/actions/Report.ts

```diff
@@ -2941,13 +2983,286 @@ function getReportPrivateNote(reportID: string | undefined) {
+        const completedTaskReportAction = task.autoCompleted
+            ? ReportUtils.buildOptimisticTaskReportAction(currentTask.reportID, CONST.REPORT.ACTIONS.TYPE.TASK_COMPLETED, 'marked as complete', actorAccountID, 2)
+            : null;
+
+        return {
+            task,
+            currentTask,
+            taskCreatedAction,
+            taskReportAction,
+            subtitleComment,
+            instructionComment,
+            completedTaskReportAction,
+        };
+    });
+
+    const tasksForParameters = tasksData.reduce<TaskForParameters[]>(
+        (acc, {task, currentTask, taskCreatedAction, taskReportAction, subtitleComment, instructionComment, completedTaskReportAction}) => {
+            const instructionCommentAction: OptimisticAddCommentReportAction = instructionComment.reportAction;
+            const instructionCommentText = instructionComment.commentText;
+            const instructionMessage: TaskMessage = {
+                reportID: currentTask.reportID,
+                reportActionID: instructionCommentAction.reportActionID,
+                reportComment: instructionCommentText,
+            };
+
+            const tasksForParametersAcc: TaskForParameters[] = [
+                ...acc,
+                {
+                    type: 'task',
+                    task: task.type,
+                    taskReportID: currentTask.reportID,
+                    parentReportID: currentTask.parentReportID ?? '',
+                    parentReportActionID: taskReportAction.reportAction.reportActionID,
+                    assigneeChatReportID: '',
+                    createdTaskReportActionID: taskCreatedAction.reportActionID,
+                    completedTaskReportActionID: completedTaskReportAction?.reportActionID ?? undefined,
+                    title: currentTask.reportName ?? '',
+                    description: currentTask.description ?? '',
+                },
+                {
+                    type: 'message',
+                    ...instructionMessage,
+                },
+            ];
+
+            if (subtitleComment) {
+                const subtitleCommentAction: OptimisticAddCommentReportAction = subtitleComment.reportAction;
+                const subtitleCommentText = subtitleComment.commentText;
+                const subtitleMessage: TaskMessage = {
+                    reportID: currentTask.reportID,
+                    reportActionID: subtitleCommentAction.reportActionID,
+                    reportComment: subtitleCommentText,
+                };
+
+                tasksForParametersAcc.push({
+                    type: 'message',
+                    ...subtitleMessage,
+                });
+            }
+
+            return tasksForParametersAcc;
+        },
+        [],
+    );
+
+    const tasksForOptimisticData = tasksData.reduce<OnyxUpdate[]>(
```

Comment:
I think we should add more plain english comments into this method as there is lots of things being created here and it can see very random unless someone is familiar with this flow

Reply:
I think is best to leave this for the future